lib/pure/asyncdispatch.nim

```diff
@@ -640,45 +635,15 @@ when defined(windows) or defined(nimdoc):
 
     let ret = WSARecv(socket.SocketHandle, addr dataBuf, 1, addr bytesReceived,
                       addr flagsio, cast[POVERLAPPED](ol), nil)
-    if ret == -1:
+    # Processing of "immediate completion" operations was removed, because under
+    # heavy load it can lead to deep recursion and stack overflow.
+    # This change also dont affect speed.
+    if ret != 0:
       let err = osLastError()
       if err.int32 != ERROR_IO_PENDING:
-        if dataBuf.buf != nil:
-          dealloc dataBuf.buf
-          dataBuf.buf = nil
+        dealloc dataBuf.buf
         GC_unref(ol)
-        if flags.isDisconnectionError(err):
```

Comment:
I don't think it's that easy to reproduce it. You've created a simple clean disconnection, but the ``isDisconnectionError`` handles many types of errors. I can't let you simply remove it.